build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/release/PruneChangelogsTask.java

```diff
@@ -0,0 +1,185 @@
+
+import org.elasticsearch.gradle.VersionProperties;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.process.ExecOperations;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+
+/**
+ * Once a minor release has happened, we no longer need to keep the changelog files that went into
+ * that release in the development branch for that major series or the branch for the next major
+ * series
+ * <p>
+ * This last examines the git history in order to work out which files can be deleted, and
+ * does the deletion after confirming with the user.
+ */
+public class PruneChangelogsTask extends DefaultTask {
+    private static final Logger LOGGER = Logging.getLogger(PruneChangelogsTask.class);
+
+    private final ConfigurableFileCollection changelogs;
+    private final GitWrapper gitWrapper;
+    private final Path rootDir;
+
+    private boolean confirmed;
+
+    @Inject
+    public PruneChangelogsTask(Project project, ObjectFactory objectFactory, ExecOperations execOperations) {
+        changelogs = objectFactory.fileCollection();
+        gitWrapper = new GitWrapper(execOperations);
+        rootDir = project.getRootDir().toPath();
+    }
+
+    @InputFiles
+    public FileCollection getChangelogs() {
+        return changelogs;
+    }
+
+    public void setChangelogs(FileCollection files) {
+        this.changelogs.setFrom(files);
+    }
+
+    @Input
+    public boolean isConfirmed() {
+        return confirmed;
+    }
+
+    @Option(option = "confirm", description = "Required in order to actually remove files")
+    public void setConfirmed(boolean confirmed) {
+        this.confirmed = confirmed;
+    }
+
+    @TaskAction
+    public void executeTask() {
+        findAndDeleteFiles(
+            this.gitWrapper,
+            files -> files.stream().filter(each -> each.delete() == false).collect(Collectors.toSet()),
+            QualifiedVersion.of(VersionProperties.getElasticsearch()),
+            this.getChangelogs().getFiles(),
+            this.confirmed,
+            this.rootDir
+        );
+    }
+
+    @VisibleForTesting
+    static void findAndDeleteFiles(
+        GitWrapper gitWrapper,
+        DeleteHelper deleteHelper,
+        QualifiedVersion version,
+        Set<File> allFilesInCheckout,
+        boolean confirmed,
+        Path rootDir
+    ) {
+        if (allFilesInCheckout.isEmpty()) {
```

Comment:
We can also add `@SkipWhenEmpty` to `getChangeLogs()` which will give us the same result.